engine/schema/src/com/cloud/network/dao/LoadBalancerDao.java

```diff
@@ -29,4 +30,6 @@
 
     List<LoadBalancerVO> listInTransitionStateByNetworkIdAndScheme(long networkId, Scheme scheme);
 
+    boolean isLoadBalancerRulesMappedToVmGuestIp(long instanceId, String instanceIp, long networkId, FirewallRule.Purpose purpose);
```

Comment:
@SudharmaJain : As per suggestion, removed Purpose parameter from method signature and added directly in the SearchBuilder. 

Please check.